sources/rasterio/large_image_source_rasterio/__init__.py

```diff
@@ -0,0 +1,1078 @@
+#############################################################################
+#  Copyright Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#############################################################################
+
+import math
+import pathlib
+import tempfile
+import threading
+from contextlib import suppress
+
+import numpy as np
+import PIL.Image
+import rasterio as rio
+from affine import Affine
+from pyproj import CRS, Transformer
+from pyproj.exceptions import CRSError
+from rasterio.enums import ColorInterp, Resampling
+from rasterio.errors import RasterioIOError
+from rasterio.warp import calculate_default_transform, reproject
+from rasterio.windows import from_bounds
+
+import large_image
+from large_image.cache_util import CacheProperties, LruCacheMetaclass, methodcache
+from large_image.constants import (TILE_FORMAT_IMAGE, TILE_FORMAT_NUMPY,
+                                   TILE_FORMAT_PIL, SourcePriority,
+                                   TileInputUnits, TileOutputMimeTypes)
+from large_image.exceptions import (TileSourceError,
+                                    TileSourceFileNotFoundError,
+                                    TileSourceInefficientError)
+from large_image.tilesource.geo import GDALBaseFileTileSource
+
+try:
+    from importlib.metadata import PackageNotFoundError
+    from importlib.metadata import version as _importlib_version
+except ImportError:
+    from importlib_metadata import PackageNotFoundError
+    from importlib_metadata import version as _importlib_version
+try:
+    __version__ = _importlib_version(__name__)
+except PackageNotFoundError:
+    # package is not installed
+    pass
+
```

Comment:
We want to filter out some additional warnings (or suppress them just where they occur), probably with 
`warnings.filterwarnings('ignore', category=rio.errors.NotGeoreferencedWarning, module='rasterio')`